src/main/java/nextstep/subway/Fare/domain/Fare.java

```diff
@@ -0,0 +1,26 @@
+package nextstep.subway.Fare.domain;
+
+public class Fare {
+    private static final int DISCOUNT_FARE = 350;
+    private int value;
+
+    public Fare(int value) {
+        this.value = value;
+    }
+
+    public int getValue() {
+        return value;
+    }
+
+    public Fare plus(Fare fare) {
+        return new Fare(value + fare.getValue());
+    }
+
+    public Fare discountChild() {
+        return new Fare((int) ((value - DISCOUNT_FARE) * 0.5));
+    }
+
+    public Fare discountTeenager() {
+        return new Fare((int) ((value - DISCOUNT_FARE) * 0.8));
+    }
```

Comment:
코드가 다소 비슷해보여서 천천히 보니 
`return new Fare((int) ((변하지않는부분 - 변하는부분1) * 변하는부분2));`
이렇게 보였어요 이렇게 표현한다면 두개의 함수가 같은 함수 같더라구요.
그래서 변하는부분을 인자로 받을수 있도록 접근해보았어요.
```suggestion
    public Fare discountFare(discount, rate) {
        return new Fare((int) ((value - discount) * rate));
    }
```

그렇게 메서드를 조정해서 다시 보니 
`DISCOUNT_FARE`이 `Fare`객체에서 가지고 있는것이 맞는가? 생각이 들어요!

이러한 관점에 대해서 어떻게 생각하시나요!? :thinking:

Reply:
`Age` 를 파라미터로 받아서 계산하도록 개선하였습니다!